estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,93 @@
+from odoo import fields, models, api
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from odoo.exceptions import UserError, ValidationError
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Property Offers for Real Estate App"
+    _order = "price desc"
+
+    price = fields.Float(string="Price")
+    status = fields.Selection(
+        selection=[
+            ("accepted", "Accepted"),
+            ("refused", "Refused"),
+        ],
+        copy=False,
+        string="Status",
+    )
+    partner_id = fields.Many2one("res.partner", string="Partner", required=True)
+    property_id = fields.Many2one("estate.property", string="Property", required=True)
+    validity = fields.Integer(default=7)
+    date_deadline = fields.Date(
+        compute="_compute_date_deadline", inverse="_inverse_date_deadline"
+    )
+    property_type_id = fields.Many2one(
+        related="property_id.property_type_id", store=True
+    )
+
+    _sql_constraints = [
+        (
+            "check_offer_price",
+            "CHECK(price > 0)",
+            "Offer Price for the property should be strictly positive",
+        )
+    ]
+
+    @api.depends("validity")
+    def _compute_date_deadline(self):
+        for record in self:
+            if record.create_date:
+                record.date_deadline = record.create_date.date() + relativedelta(
+                    days=record.validity
+                )
+            else:
+                record.date_deadline = datetime.today() + relativedelta(
+                    days=record.validity
+                )
+
+    def _inverse_date_deadline(self):
+        for record in self:
+            if record.date_deadline:
+                delta = record.date_deadline - datetime.today().date()
```

Comment:
```suggestion
                delta = record.date_deadline - datetime.today().date()
```
datetime.today() why always today and not create date

Reply:
Firstly I was thinking like when we are creating the offer and if we want to compute the validity from the deadline, the deadline is subtracted from the day when the record is created i.e we are creating the offer record at the current time but after that i realized my mistake that what if I create the offer record but not given the validity and deadline and we have to calculate on the basis of create date priority as per the exercise. 

We can both the methods like these:
`@api.depends("validity")
    def _compute_date_deadline(self):
        for record in self:
            create_date = (
                record.create_date.date() if record.create_date else fields.Date.today()
            )
            record.date_deadline = create_date + timedelta(days=record.validity)

    def _inverse_date_deadline(self):
        for record in self:
            if record.date_deadline:
                create_date = (
                    record.create_date.date() if record.create_date else fields.Date.today()
                )
                record.validity = (record.date_deadline - create_date).days`